packages/FileFormatter/FileFormatter.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\FileFormatter;
+
+use Rector\Core\Configuration\Option;
+use Rector\Core\Contract\EditorConfig\EditorConfigParserInterface;
+use Rector\Core\Contract\Formatter\FileFormatterInterface;
+use Rector\Core\ValueObject\Application\File;
+use Symplify\PackageBuilder\Parameter\ParameterProvider;
+
+final class FileFormatter
+{
+    /**
+     * @var EditorConfigParserInterface
+     */
+    private $editorConfigParser;
+
+    /**
+     * @var FileFormatterInterface[]
+     */
+    private $fileFormatters;
+
+    /**
+     * @var ParameterProvider
+     */
+    private $parameterProvider;
+
+    /**
+     * @param FileFormatterInterface[] $fileFormatters
+     */
+    public function __construct(EditorConfigParserInterface $editorConfigParser, ParameterProvider $parameterProvider, array $fileFormatters = [])
+    {
+        $this->editorConfigParser = $editorConfigParser;
+        $this->fileFormatters = $fileFormatters;
+        $this->parameterProvider = $parameterProvider;
+    }
+
+    /**
+     * @param File[] $files
+     */
+    public function format(array $files): void
+    {
+        if(!$this->parameterProvider->provideBoolParameter(Option::ENABLE_EDITORCONFIG)) {
+            return;
+        }
+
+        foreach ($files as $file) {
+            if (! $file->hasChanged()) {
```

Comment:
Why only changed files? :) If I want to replace all tabs to 4 spaces, I have to change all files first... If I enable formatting, I want to change all files

Reply:
That's out of our scope. But phpstorm can do that, or editor config docker image ;)

Reply:
Well, PHPStorm can also rename classes, methods and namespaces ;) It's just another tool you have to combine with Rector

Reply:
The main reason why i implemented this feature was the fact that many people were complaining that touched files like composer.json are reformatted by rector. And i came up with the idea to take the editorconfig into account. I don´t want to touch all scanned files of the project. Only those touched by rector.

Reply:
OK :)